### PR TITLE
Implement UI checkboxes for slot toggles: locked, shareable, sublevel, lbp1only.

### PR DESCRIPTION
**Pull request by FeTetra**
This PR adds some simple UI into the slot settings page for enabling some options (as listed in the title), which are typically accessible in-game.

### CODE REVIEW

File: ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml [hunk 1]
```diff
@@ -52,6 +52,37 @@ function onSubmit(){
                         <input type="hidden" name="avatar" id="avatar-encoded">
                     </div>
                 </div>
+
+                <div class="ui divider"></div>
+                <div class="ui red label">
+                    <i class="lock icon"></i>
+                    Locked
```
Comment (Slendy) [inline ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml:59]:
```suggestion
                    <label for="locked">
                        <i class="lock icon"></i>
                        Locked
                    </label>
```
By wrapping this in a label with a `for` attribute you can allow the checkbox to be clickable from almost anywhere in the box rather than the checkbox itself

File: ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml [hunk 2]
```diff
@@ -52,6 +52,37 @@ function onSubmit(){
...
+                    <input type="checkbox" name="locked" id="locked" style="margin-left: 5px;">
+                </div>
+                <div class="ui green label">
+                    <i class="check icon"></i>
+                    Copyable
```
Comment (Slendy) [inline ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml:64]:
```suggestion
                    <label for="copyable">
                        <i class="icon copy outline"></i>
                        Copyable
                    </label>
```
Same as above. I think the copyable button having a checkmark is a little ambiguous, somehting like a traditional copy icon might work better.

File: ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml [hunk 3]
```diff
@@ -52,6 +52,37 @@ function onSubmit(){
...
+                    <input type="checkbox" name="copyable" id="copyable" style="margin-left: 5px;">
+                </div>
+                @if (Model.Slot.GameVersion != GameVersion.LittleBigPlanet1)
+                {
+                    <div class="ui blue label">
+                        <i class="arrow circle down icon"></i>
+                        Sub Level
```
Comment (Slendy) [inline ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml:71]:
```suggestion
                        <label for="subLevel">
                            <i class="arrow circle down icon"></i>
                            Sub Level
                        </label>
```

File: ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml.cs [hunk 4]
```diff
@@ -50,11 +50,15 @@ public async Task<IActionResult> OnPost([FromRoute] int slotId, [FromForm] strin
 
         if (labels != null)
         {
-            labels = LabelHelper.RemoveInvalidLabels(labels);
```
Comment (Slendy) [inline ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml.cs:53]: Why was this removed?

File: ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml [hunk 5]
```diff
@@ -92,78 +122,78 @@ function onSubmit(){
...
+        return;
+    
+    const output = document.getElementById('slotIcon');
+    const reader = new FileReader();
+    reader.onload = function(){
+        output.src = reader.result;
+        selectedAvatar = reader.result;
+    };
+    reader.readAsDataURL(file);
+}
```
Comment (Slendy) [inline ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml:198]: Can we revert the indentation to how it was before

File: ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml.cs [hunk 6]
```diff
@@ -50,11 +50,15 @@ public async Task<IActionResult> OnPost([FromRoute] int slotId, [FromForm] strin
...
             if (this.Slot.AuthorLabels != labels)
                 this.Slot.AuthorLabels = labels;
         }
 
+        if (this.Slot.InitiallyLocked != locked) this.Slot.InitiallyLocked = locked;
+        if (this.Slot.Shareable != (copyable ? 1 : 0)) this.Slot.Shareable = (copyable ? 1 : 0);
```
Comment (Slendy) [inline ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml.cs:58]:
```suggestion
        if (this.Slot.Shareable != Convert.ToInt32(copyable)) this.Slot.Shareable = Convert.ToInt32(copyable);
```

File: ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml.cs [hunk 7]
```diff
@@ -50,11 +50,15 @@ public async Task<IActionResult> OnPost([FromRoute] int slotId, [FromForm] strin
...
+        if (this.Slot.SubLevel != subLevel) this.Slot.SubLevel = subLevel;
+        if (this.Slot.Lbp1Only != lbp1Only) this.Slot.Lbp1Only = lbp1Only;
```
Comment (Slendy) [inline ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml.cs:60]: These should probably have slot game version checks

File: ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml [hunk 8]
```diff
@@ -52,6 +52,37 @@ function onSubmit(){
...
+                {
+                    <div class="ui yellow label">
+                        <i class="eye icon"></i>
+                        LBP1 Only
+                        <input type="checkbox" name="lbp1Only" id="lbp1Only" style="margin-left: 5px;">
+                    </div>
+                }
+                <div class="ui divider fitted hidden"></div>
+
+
```
Comment (Slendy) [inline ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml:85]:
```suggestion
```

**Review by Slendy — CHANGES_REQUESTED** [review]
First of all, thank you for your contribution to Lighthouse. I have a couple concerns about formatting and some accessibility issues with the checkboxes and labels. For functionality, the checkboxes also need an extra attribute to send `true` instead of `on` in the form so the values are properly parsed by ASP.NET. Keep in mind these suggested changes are usually done in my browser by hand so they aren't always directly committable and are more to give you an idea of what to change. Also, feel free to ask questions about my suggested changes in the comments or contest them if you think that my suggestion isn't the best way to do something.

File: ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml (hunk 8 above)
Comment (FeTetra) [reply ProjectLighthouse.Servers.Website/Pages/SlotSettingsPage.cshtml:85]: :disaster:

**Comment by FeTetra** [comment]
I will fix these issues in a new PR to clean this up, since I accidentally proceeded to commit these changes.
